src/client/components/pages/parts/delete-or-remove-collaboration-modal.js

```diff
@@ -0,0 +1,128 @@
+import * as bootstrap from 'react-bootstrap';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import PropTypes from 'prop-types';
+import React from 'react';
+import request from 'superagent';
+
+
+const {Alert, Button, Modal} = bootstrap;
+
+class DeleteOrRemoveCollaborationModal extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			error: null
+		};
+
+		this.handleSubmit = this.handleSubmit.bind(this);
+	}
+
+	handleSubmit() {
+		request.post(this.postUrl)
+			.send(this.postData)
+			.then((res) => {
+				if (this.props.isDelete) {
+					window.location.href = `/editor/${this.props.collection.ownerId}/collections`;
+				}
+				else {
+					window.location.href = `/editor/${this.props.userId}/collections`;
+				}
+			}, (error) => {
+				this.setState({
+					error: 'Something went wrong! Please try again later'
+				});
+			});
+	}
+
+	render() {
+		const {collection} = this.props;
+		// eslint-disable-next-line one-var
+		let modalBody, modalTitle, submitButton;
+		if (this.props.isDelete) {
+			this.postUrl = `/collection/${collection.id}/delete/handler`;
+			this.postData = {};
+			modalTitle = 'Confirm deletion';
+			modalBody = (
+				<Alert bsStyle="warning">
+					<h4>
+						<FontAwesomeIcon icon="exclamation-triangle"/>&nbsp;
+						You’re about to delete the Collection: {collection.name}.
+					</h4>
+					<p>
+						Make sure you actually want to delete this Collection <br/>
+						There is no way to undo this.
+					</p>
+				</Alert>
+			);
+			submitButton = (
+				<Button bsStyle="danger" onClick={this.handleSubmit}>
+					<FontAwesomeIcon icon="trash-alt"/> Delete
+				</Button>
+			);
+		}
+		else {
+			// loggedInUser must be collaborator here
+			this.postUrl = `/collection/${collection.id}/collaborator/remove`;
+			this.postData = {collaboratorIds: [this.props.userId]};
+			modalTitle = 'Remove yourself as a collaborator';
+			modalBody = (
+				<Alert bsStyle="warning">
+					<h4>
+						<FontAwesomeIcon icon="exclamation-triangle"/>&nbsp;
+						You’re about to remove yourself as a collaborator of Collection: {collection.name}.
+					</h4>
+					<p>
+						Are you sure you want to proceed ? You won’t be able to undo this.
```

Comment:
```suggestion
						Are you sure you want to proceed ? You won’t be able to undo this yourself and will need to ask the collection's owner to add you again.
```
Nothing big, but a suggestion